towhee/functional/data_collection.py

```diff
@@ -199,21 +175,30 @@ def is_stream(self):
         """
         return isinstance(self._iterable, Iterator)
 
-    @property
-    def _factory(self):
+    def _factory(self, iterable, parent_stream = True):
         """
         Factory method for data collection.
 
-        This factory method has been wrapped into a `param_scope()` which contains parent infomations.
-        """
-        creator = DataCollection.stream if self.is_stream else DataCollection.unstream
+        This factory method has been wrapped into a `param_scope()` which contains parent information.
 
-        def wrapper(*arg, **kws):
-            with param_scope() as hp:
-                hp().data_collection.parent = self
-                return creator(*arg, **kws)
+        Args:
+            iterable: An iterable object, the data being stored in the DC
+            parent_stream: Whether to copy the parents format (streamed vs unstreamed)
+
+        Returns:
+            DataCollection: DataCollection encapsulating the iterable.
+        """
+        if parent_stream is True:
```

Comment:
Do we have to pass this parameter `parent_stream` on every _factory call?

In most cases, the _factoru receives a generator produced by mapping an inner function to the iterable. In such actions, DC always returns a stream child with the default `parent_stream `, but we expect an unstreamed one if the parent is unstreamed.

Reply:
And there lacks a test case for checking this behavoir, stream in stream out, unstream in unstream out.